core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -266,6 +266,11 @@ object SparkSubmit {
       }
     }
 
+    // SPARK-5966, check deployMode CLUSTER and master local
+    if (clusterManager == LOCAL && deployMode == CLUSTER) {
+      printErrorAndExit("Cluster deploy mode is not compatible with master \"local\"")
+    }
```

Comment:
Hello Andrew: Thanks for pointing this out, I have made the code changes, run the test , and submit the pull request, can you help review? 
Kevin